core/src/main/java/com/expedia/adaptivealerting/core/anomaly/AnomalyResult.java

```diff
@@ -102,7 +113,15 @@ public AnomalyLevel getAnomalyLevel() {
     public void setAnomalyLevel(AnomalyLevel anomalyLevel) {
         this.anomalyLevel = anomalyLevel;
     }
-    
+
+    public String getDetector() {
+        return detector;
+    }
+
+    public void setDetector(String detector) {
```

Comment:
Wouldn't it be better to just get/set the Class itself? The worry is this. When somebody creates an AnomalyResult(Class), we enforce a classname for the detector. That works for me. But then somebody can come in later and call `setDetector("im-no-class")`. Either we care about the detector being a class or not. If so, let's use the class. If not, then let's change the constructor.